packages/js/src/utils/package-manager-workspaces.ts

```diff
@@ -58,7 +61,22 @@ export function isWorkspacesEnabled(
   return false;
 }
 
+function detectPackageManager(tree: Tree, dir: string = ''): PackageManager {
```

Comment:
Why was this inlined?

Reply:
The one from devkit is not tree-based. It works with the filesystem directly, but these are used in a generator context, which is tree-based.

Reply:
I appreciate what you are saying, but I imagine the reason this hasn't come up before is that the distinction isn't important, because the package manager (proxied via the lock file name) simply is not likely at all to change during the execution of a generator (and it certainly wouldn't during the ones in question here)

Reply:
I just don't love that this would be the 3rd time we have a detectPackageManager function and none of them share any logic...

Reply:
Yeah, that makes sense. I'll revert it.

Reply:
In CNW with TS + PM Workspaces - will this ever be executed when there are _no_ files persisted to disk already?

I don’t think so because we always create the sandbox first before running generators. 

However, having one that works with a tree might be useful for anyone Extending Nx with custom plugins, generators, create-package etc. 

Just not sure if it would be in the scope of this PR

Reply:
When the lock file doesn't exist on disk, the install package tasks would use the existing `detectPackageManager` function, so using it inside the generators before the file exists yields the same result. That's why, as James mentioned, it should be fine in most cases, if not all.

Reply:
@JamesHenry PR is updated now.